2017/update password

dominikx96 opened the pull request:
# Pull Request Template

## Issue Overview

This PR addresses #2017 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Supports an error handling for the outdated password. Also, forgot-password pages have been refactored to use reusable form.

## How Can This Be Tested/Reviewed?

1. Reseed a database
2. Use `user+expired@example.com`  / `abcdef` to log in
3. You should see an error alert with a link to the forgot-password page
4. Try to reset password and log in again

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

jaredcwhite reviewed (changes requested):
Hey @dominikx96, a couple of things I noticed:

* We need to keep anything Next.js-specific out of `ui-components` at this point. So for the new page components, either they should use the `router` variable from `useContext(NavigationContext)` or accept a prop function
* I'm not sure having embedded and dangerously setting HTML is the best approach on the alert messaging. Maybe there's a way to refactor that, or alternatively use Markdown

dominikx96 commented:
> Hey @dominikx96, a couple of things I noticed:
> 
> * We need to keep anything Next.js-specific out of `ui-components` at this point. So for the new page components, either they should use the `router` variable from `useContext(NavigationContext)` or accept a prop function
> * I'm not sure having embedded and dangerously setting HTML is the best approach on the alert messaging. Maybe there's a way to refactor that, or alternatively use Markdown

Both points make sense to me, thanks @jaredcwhite! I've updated the alert text to use markdown and removed Nextjs dependencies.

seanmalbert reviewed (approved):
This looks good to me, @dominikx96. Can you please resolve the merge conflict. @jaredcwhite , can you take another look at this with the updates from your comments.

jaredcwhite commented on ui-components/src/config/NavigationContext.tsx:
```diff
@@ -23,6 +23,7 @@ export interface GenericRouterOptions {
 
 export interface GenericRouter {
   push: (url: Url, as?: Url, options?: GenericRouterOptions) => void
+  back(): void
```
I think this should be `back: () => void`

jaredcwhite reviewed (approved):
Just saw one little thing, otherwise LGTM :+1:
